lambdas/service/app.py

```diff
@@ -546,204 +546,244 @@ def repository_search(entity_type: str, item_id: str):
         raise NotFoundError(msg=e)
 
 
-@app.route('/repository/files', methods=['HEAD', 'GET'], cors=True)
-@app.route('/repository/files/{file_id}', methods=['GET'], cors=True)
+generic_dict = schema.object(additional_properties=True)
+array_of_dict = schema.array(generic_dict)
+hit_schema = schema.object(
+    additional_properties=True,
+    protocols=array_of_dict,
+    entryId=str,
+    samples=array_of_dict,
+    specimens=array_of_dict,
+    cellLines=array_of_dict,
+    donorOrganisms=array_of_dict,
+    organoids=schema.array(str),
+    cellSuspensions=array_of_dict,
+)
+
+page_schema = schema.object(
+    hits=schema.array(hit_schema),
+    pagination=generic_dict,
+    termFacets=generic_dict
+)
+
+
+def repository_search_spec(entity_type):
+    return {
+        'tags': ['Repository'],
+        'parameters': [
+            params.query('filters',
+                         schema.optional(str),
+                         description='Filters to be applied when calling ElasticSearch.'),
```

Comment:
This is disappointingly brief (if you were a reader, unfamiliar with the API, would you know what to pass for `filters` given this documentation?) and incorrect. The `filter` parameter is a JSON structure and that needs to be expressed somehow. Either formally via JSONSchema or OpenAPI or, if that's technically not possible, verbally.